src/Elastic.Transport/Components/Serialization/SystemTextJsonSerializer.cs

```diff
@@ -84,6 +84,29 @@ public override Task SerializeAsync<T>(T data, Stream stream,
 
 	#endregion Serializer
 
+	/// <summary>
+	/// Override to (dis-)allow fast-path (de-)serialization for specific types.
+	/// </summary>
+	/// <param name="type">The <see cref="Type"/> that is being (de-)serialized.</param>
+	/// <returns>
+	/// <see langword="true"/> if the given <paramref name="type"/> supports fast-path (de-)serialization or
+	/// <see langword="false"/>, if not.
+	/// </returns>
+	/// <remarks>
+	/// <para>
+	///	Most extension methods in <see cref="Extensions.TransportSerializerExtensions"/> will prefer fast-path (de-)serialization, when
+	/// used with <see cref="SystemTextJsonSerializer"/> based serializer implementations.
+	/// Fast-path (de-)serialization bypasses the <see cref="Deserialize"/>, <see cref="Deserialize{T}"/>, <see cref="Serialize{T}"/>
+	/// methods and directly uses the <see cref="JsonSerializer"/> API instead.
+	/// </para>
+	/// <para>
+	///	In some cases, when the concrete <see cref="SystemTextJsonSerializer"/> based serializer implementation overrides one or more of
+	/// the named methods, the default fast-path behavior might be undesired. The <see cref="SupportsFastPath"/> method can be used to
```

Comment:
Can you expand why the fast path might be undesirable?

I grasp the use case for not prefering the fast path.

Reply:
If the user overrides one of these methods, there is probably a reason for doing so.

Like mentioned, fast path will bypass the Deserialize/Serialize method invocation, which in conclusion will as well bypass execution of the user code (which is why fast path is undesirable in these cases, otherwise there wouldn't have been a good reason for overriding Deserialize/Serialize in the first place).

Is that unclear from the doc comment?

Reply:
Comments make more sense in the morning then late at night :)

Reply:
True :D Thanks for the review! I will try to rephrase it a bit before merging to make it more clear.